ssas/systems.go

```diff
@@ -414,6 +431,41 @@ func GenerateSecret() (string, error) {
 	return fmt.Sprintf("%x", b), nil
 }
 
+/*
+	ResetSecret creates a new secret for the current system.
+*/
+func (system *System) ResetSecret(trackingID string) (string, error) {
+	db := GetGORMDbConnection()
+	defer Close(db)
+
+	newSecretEvent := Event{Op: "ResetSecret", TrackingID: trackingID, ClientID: system.ClientID}
+	OperationStarted(newSecretEvent)
+
+	secretString, err := GenerateSecret()
+	if err != nil {
+		newSecretEvent.Help = fmt.Sprintf("could not reset secret for clientID %s: %s", system.ClientID, err.Error())
+		OperationFailed(newSecretEvent)
+		return "", errors.New("internal system error")
```

Comment:
👍 on returning a more generic error message, so that future public-facing endpoints can safely pass along the message.